### ISSUE TITLE
Unable to resolve dependencies with k8s/client-go and kubevirt/client-go

### ISSUE THREAD

**Issue by j-griffith** [kubevirt/kubevirt#2919]
/kind bug

**What happened**:

While trying to consume kubevirt/client-go from a project generated by kubebuilder, I ran into what seem to be unresolvable conflicts with gos packaging of the k8s/client-go.

The simple example provided here: https://github.com/kubevirt/client-go/tree/master/examples/listvms works fine.  However, when trying to consume from something that uses a newer version of k8s.io/client-go I can't seem to resolve the dependencies. 

The problem is that v1alpha1 has been removed from k8s.io/client-go, and kubebuilder requires newer versions that use v1.  I've tried to mess with the go.mod file to make this work:

```
go 1.13
require (
	github.com/go-logr/logr v0.1.0
	github.com/onsi/ginkgo v1.8.0
	github.com/onsi/gomega v1.5.1-0.20190515112211-6a48b4839f85
	k8s.io/apimachinery v0.0.0-20190913080033-27d36303b655
	k8s.io/client-go v11.0.0+incompatible
	kubevirt.io/client-go v0.23.0
	sigs.k8s.io/controller-runtime v0.4.0
)
```

but that doesn't seem to resolve the two independently and I get this failure during build:

`../../../../pkg/mod/kubevirt.io/client-go@v0.19.0/kubecli/generated_mock_kubevirt.go:17:2: unknown import path "k8s.io/client-go/kubernetes/typed/admissionregistration/v1alpha1": cannot find module providing package k8s.io/client-go/kubernetes/typed/admissionregistration/v1alpha1
`

To be fair I don't think this is really a "kubevirt" issue, but more a go dependency and packaging issue, but perhaps there's a clever way to address it in kubevirt.  As it stands right now it doesn't seem possible to consume kubevirt client-go from non trivial projects.

**What you expected to happen**:

There should be an easy way for me to consume the kubevirt/go-client from a kubebuilder package, even if that just means we need to bump kubevirt/go-client to a newer version of k8s.io/client-go

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a project with kubebuilder
2. Add some implementation in a controller that uses kubevirt/client-go

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by cynepco3hahue** [comment]
probably we need to update KubeVirt to use k8s-1.16

**Comment by j-griffith** [comment]
> probably we need to update KubeVirt to use k8s-1.16

I think that would defintely help, I tried to hack my own package with that but didn't have much success.  If it helps I've provided a bit more concrete info here:  https://github.com/j-griffith/cluster-api-provider-kubevirt/commit/a7a43fe0e676ef8f01a1a87413bf28d52e5e37d8

This strips everything down to the issue with the import of kubevirt/client-go and the commit comment includes some info about what I'm seeing.  There may be an easy way to solve this with go.mod directives, but I'm afraid I can't figure it out.
